src/monmove.cpp

```diff
@@ -996,6 +997,108 @@ bool monster::move_to( const tripoint &p, bool force )
     return true;
 }
 
+bool monster::push_to( const tripoint &p, const int boost, const size_t depth )
+{
+    if( !has_flag( MF_PUSH_MON ) || depth > 2 || has_effect( "pushed" ) ) {
+        return false;
+    }
+
+    // TODO: Generalize this to Creature
+    const int mondex = g->mon_at( p );
+    if( mondex < 0 ) {
+        return false;
+    }
+
+    monster *critter = &g->zombie( mondex );
+    if( critter == nullptr || critter == this || p == pos() ) {
```

Comment:
Any specific reasons why `critter` is a pointer and not a reference? Note that `critter` can not be NULL, `game::zombie` returns a reference and references can refer to NULL.